### PR TITLE
Add optional parameters fft_based_translation_initialization

### PR DESCRIPTION
**Pull request by blowekamp**
Add required_fraction_of_overlapping_pixels to specify minimum overlapping of images. Defaulting to 0 to preserve compatibility.

Add initial_transform and masked_pixel_value.

### CODE REVIEW

File: SimpleITK/utilities/fft.py [hunk 1]
```diff
@@ -20,7 +20,12 @@
 
 
 def fft_based_translation_initialization(
-    fixed: sitk.Image, moving: sitk.Image
+    fixed: sitk.Image,
+    moving: sitk.Image,
+    *,
+    required_fraction_of_overlapping_pixels: float = 0.0,
+    initial_transform: sitk.Transform = None,
+    maked_pixel_value: float = None,
```
Comment (zivy) [inline SimpleITK/utilities/fft.py:28]: missing s `masked_pixel...` not `maked_pixel...` auto complete ensured the spelling mistake was consistent in the documentation and code, so requested change is not limited to this line.

File: SimpleITK/utilities/fft.py [hunk 2]
```diff
@@ -34,30 +39,61 @@ def fft_based_translation_initialization(
...
-
-    sigma = fixed.GetSpacing()[0]
-    pixel_type = sitk.sitkFloat32
 
-    fft_fixed = sitk.Cast(sitk.SmoothingRecursiveGaussian(fixed, sigma), pixel_type)
-    fft_moving = sitk.Cast(sitk.SmoothingRecursiveGaussian(moving, sigma), pixel_type)
+        # anti_aliasing_sigma = [
+        #    max((nspc - ospc) / 2.0, 0.0)
+        #    for ospc, nspc in zip(moving.GetSpacing(), fixed.GetSpacing())
+        # ]
```
Comment (zivy) [inline SimpleITK/utilities/fft.py:64]: Please remove the commented out code.

**Comment by blowekamp** [comment]
@zivy  This PR changes the required version of SimpleITK to 2.3.0 for the number of required pixels overlap parameter for the FFTNormalizedCorelation. Do you thing it is ok to update the require version of SimpleITK like that?

**Review by zivy — APPROVED** [review]
Minor spelling and commented out code that should be removed.

File: SimpleITK/utilities/fft.py (hunk 1 above)
Comment (blowekamp) [reply SimpleITK/utilities/fft.py:28]: O that change locally.

**Comment by zivy** [comment]
At this early stage of the utilities package life (<v0.5) I think the change requiring SimpleITK 2.3.0 is fine.

**Pull request merged**